theano/gpuarray/blas.py

```diff
@@ -119,11 +119,11 @@ def c_code(self, node, name, inp, out, sub):
             if (%(A)s->ga.flags & GA_C_CONTIGUOUS) {
                 ssize_t a_stride0 = %(A)s->ga.strides[0];
                 %(A)s->ga.strides[0] = %(A)s->ga.strides[1];
-                if (pygpu_blas_rdot(%(x)s, %(A)s, %(y)s, 0) == -1) {
+                if (pygpu_blas_rdot(%(x)s, %(A)s, %(out)s, 0) == -1) {
```

Comment:
This would always have crashed here, since `y` had the wrong number of dimensions if it was not aliased to `out`.